lib/glific/messages.ex

```diff
@@ -759,4 +759,33 @@ defmodule Glific.Messages do
     Message
     |> struct(opts)
   end
+
+  @doc """
+  Get session uuid of contact's last inbound message
+  """
+  @spec get_session_uuid(map()) :: Ecto.UUID.t()
+  def get_session_uuid(message_params) do
+    organization = Partners.organization(message_params.organization_id)
+
+    with {:ok, contact} <-
+           Repo.fetch_by(Contact, %{phone: message_params.sender.phone}),
+         true <-
+           Timex.diff(DateTime.utc_now(), contact.last_message_at, :minutes) <
+             organization.session_limit,
+         false <-
+           is_nil(
+             last_inbound_message =
+               Message
+               |> where([m], m.contact_id == ^contact.id)
+               |> where([m], m.flow == "inbound")
+               |> order_by([m], desc: m.id)
+               |> first()
+               |> Repo.one()
```

Comment:
Why we need both functions  |> first and Repo.one ?

Reply:
We need |> first to create the query, and then Repo.one() or Repo.all() to fetch it.